src/dso_api/dynamic_api/openapi.py

```diff
@@ -104,29 +109,65 @@ def _get_openapi_view(dataset: Dataset, renderer_classes=None):
 
     # The second strategy is chosen here to keep the whole endpoint enumeration logic intact.
     # Patterns is a lazy object so it's not evaluated yet while the URLconf is being constructed.
-    view = get_schema_view(
+    openapi_view = get_schema_view(
         title=dataset_schema.title,
         description=dataset_schema.description or "",
-        renderer_classes=renderer_classes,
+        renderer_classes=[renderers.JSONOpenAPIRenderer],
         patterns=_lazy_get_dataset_patterns(dataset_schema.id),
         generator_class=DynamicApiSchemaGenerator,
         permission_classes=(permissions.AllowAny,),
         version=dataset_schema.version,
     )
 
+    # Specific data to override in the OpenAPI
+    openapi_overrides = {
+        "info": {
+            "license": dataset_schema.get("license", ""),
+        },
+        "externalDocs": {
+            "url": urljoin(
+                settings.SPECTACULAR_SETTINGS["EXTERNAL_DOCS"]["url"],  # to preserve hostname
+                f"/v1/docs/datasets/{dataset_schema.id}.html",
+            )
+        },
+    }
+
     # As get_schema_view() offers no **initkwargs to the view, it's not possible to pass
     # additional parameters to the generator/view. Instead, our schema generator is directly
     # accessed here, and the override logic is reused.
     # The schema_override needs to be copied so the class attribute is not altered globally.
-    schema_generator: DSOSchemaGenerator = view.view_initkwargs["schema_generator"]
+    schema_generator: DSOSchemaGenerator = openapi_view.view_initkwargs["schema_generator"]
     schema_generator.schema_overrides = {
-        "info": {
-            "license": dataset_schema.get("license", ""),
-        },
+        **openapi_overrides,
         **schema_generator.schema_overrides,
     }
 
-    return view
+    # Wrap the view in a "decorator" that shows the Swagger interface for browsers.
+    return _swagger_on_browser(openapi_view)
+
+
+def _swagger_on_browser(openapi_view):
+    """A 'decorator' that shows the swagger interface on browser requests.
+    This is a separate function to reduce the closure context data.
+    """
+    # The ?format=json isn't really needed, but makes the fetch/XMLHttpRequest explicit
+    # to request the OpenAPI JSON and avoids any possible browser-interaction.
+    swagger_view = DSOSwaggerView.as_view(url="?format=json")
+
+    @wraps(openapi_view)
+    def _switching_view(request):
+        is_browser = "text/html" in request.META.get("HTTP_ACCEPT", "")
```

Comment:
Last sanity check: will this work for curl & python requests?

Reply:
Yes. This is the same logic as DRF uses.